File: src/Addressing.cpp
#include "Addressing.hpp"
#include <iostream>
#include <string>
using namespace std;
Addressing::Addressing() : row(0), cell(0), storage_area("") {}

Addressing::Addressing(int row, int cell, const string& storage_area) : row(row), cell(cell), storage_area(storage_area) {}

string Addressing::getAddress() const {
    return "Стилаж " + to_string(row) + ",Полка " + to_string(cell);
}
string Addressing::getFullAddress() const {
    string s = "Зона: " + storage_area + ", " + getAddress();
    return s;
}
int Addressing::getRow() const {
    return row; 
}
int Addressing::getCell() const { 
    return cell;
}
string Addressing::getStorageArea() const {
    return storage_area;
}

File: warehouse/Addressing.java
package com.warehouse;

public class Addressing {
    private int row;
    private int cell;
    private String storageArea;

    public Addressing() {
        this.row = 0;
        this.cell = 0;
        this.storageArea = "";
    }
    
    public Addressing(int row, int cell, String storageArea) {
        this.row = row;
        this.cell = cell;
        this.storageArea = storageArea;
    }
    
    public String getAddress() {
        return "Ряд " + row + ", Ячейка " + cell;
    }
    
    public String getFullAddress() {
        return "Зона: " + storageArea + ", " + getAddress();
    }
    
    public void displayAddress() {
        System.out.println(getFullAddress());
    }
    
    public int getRow() { 
        return row; 
    }
    public int getCell() { 
        return cell; 
    }
    public String getStorageArea() {
         return storageArea; 
    }
    
    public void setRow(int row) {
         this.row = row; 
    }

    public void setCell(int cell) {
         this.cell = cell; 
    }

    public void setStorageArea(String storageArea) {
         this.storageArea = storageArea;
    }
}